include/ck_tile/ops/gemm/kernel/batched_gemm_kernel.hpp

```diff
@@ -0,0 +1,220 @@
+    }
+
+    __host__ static constexpr auto BlockSize() { return dim3(KernelBlockSize); }
+
+    struct BatchedGemmCommonKargs
+    {
+        const void* a_ptr;
+        const void* b_ptr;
+        void* c_ptr;
+        index_t k_batch;
+        index_t M;
+        index_t N;
+        index_t K;
+        index_t stride_A;
+        index_t stride_B;
+        index_t stride_C;
+        index_t batch_stride_A;
+        index_t batch_stride_B;
+        index_t batch_stride_C;
+        index_t batch_count;
+    };
+
+    CK_TILE_HOST static constexpr BatchedGemmCommonKargs MakeKargs(const void* a_ptr,
+                                                                   const void* b_ptr,
+                                                                   void* c_ptr,
+                                                                   index_t k_batch,
+                                                                   index_t M,
+                                                                   index_t N,
+                                                                   index_t K,
+                                                                   index_t stride_A,
+                                                                   index_t stride_B,
+                                                                   index_t stride_C,
+                                                                   index_t batch_stride_A,
+                                                                   index_t batch_stride_B,
+                                                                   index_t batch_stride_C,
+                                                                   index_t batch_count)
+    {
+        return BatchedGemmCommonKargs{a_ptr,
+                                      b_ptr,
+                                      c_ptr,
+                                      k_batch,
+                                      M,
+                                      N,
+                                      K,
+                                      stride_A,
+                                      stride_B,
+                                      stride_C,
+                                      batch_stride_A,
+                                      batch_stride_B,
+                                      batch_stride_C,
+                                      batch_count};
+    }
+
+    CK_TILE_HOST_DEVICE static constexpr index_t GetSmemSize()
+    {
+        return max(GemmPipeline::GetSmemSize(), EpiloguePipeline::GetSmemSize());
+    }
+
+    CK_TILE_DEVICE void operator()(BatchedGemmCommonKargs kargs) const
+    {
+        const auto [i_m, i_n] = TilePartitioner{}();
+        const auto i_batch    = blockIdx.z;
+        //  options
+        const ADataType* a_start = static_cast<const ADataType*>(kargs.a_ptr) +
+                                   __builtin_amdgcn_readfirstlane(i_batch * kargs.batch_stride_A);
```

Comment:
This readfirstlane should be "outside" that is cover result of a_ptr + ibatch* stride